src/go/k8s/api/vectorized/v1alpha1/cluster_webhook_test.go

```diff
@@ -1635,6 +1635,251 @@ func TestSchemaRegistryTLSExternalCACertValidations(t *testing.T) {
 	})
 }
 
+func TestAdminAPITLSExternalCA(t *testing.T) { //nolint:funlen // better packing related tests together
+	redpandaCluster := validRedpandaCluster()
+
+	t.Run("if admin API mTLS enabled and clientCACertRef is set, name must be provided in clientCACertRef", func(t *testing.T) {
```

Comment:
nit: This looks like it could easily be refactored into a table test which would make it a lot easier to digest and maintain going forward. WDYT?